### DISCUSSION TITLE
Getting TI_norm in a volume

### DISCUSSION THREAD

**Discussion by idossha** [simnibs/simnibs#341] [8. Simulation workflows]
Hi all,

I am working on some TI simulations these days. I am wondering what will be the best way to extract the normal component of the TI field (with respect to the cortex) in a way that could be transformed to nifti (using msh2nii). I have tried coming up with something elegant based on some of the examples provided, but I did not figure it out.

Please let me know if you have some ideas. 

Thanks for all the help!
Ido

**Comment by oulap** [comment]
Hi,

Getting the normal component on a volume is a bit tricky as the normal is only defined w.r.t. a surface. What you could do is to take the central surface, which has the normal component stored in the node data, from the simulation folder and then interpolate that to a volume. The easiest way is probably to grab the coordinates of the surface nodes, transform those to voxel space using the affine matrix of the input scan, and then interpolate using e.g., scipy. If you want to do it quick-and-dirty, you can just assign the normal value to the closest voxel, i.e., do nearest neighbor interpolation.
